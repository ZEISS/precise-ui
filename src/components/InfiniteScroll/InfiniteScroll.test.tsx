import * as React from 'react';
import * as enzyme from 'enzyme';
import { InfiniteScroll } from './';

const getData = length =>
  Array.apply(undefined, { length }).map((item, index) => {
    const x = `X${index}`;
    return (
      <div key={x}>
        X:
        {x}
      </div>
    );
  });

describe('<InfiniteScroll />', () => {
  const mockCallback = jest.fn();
  it('should render an empty <InfiniteScroll>', () => {
    const wrapper = enzyme.shallow(<InfiniteScroll loadItems={mockCallback} data={[]} containerHeight={500} hasMore />);
    expect(wrapper).toMatchSnapshot();
  });

  it('should render <InfiniteScroll> with "Show More" button and data elements', () => {
    const wrapper = enzyme
      .shallow(<InfiniteScroll loadItems={mockCallback} data={getData(10)} containerHeight={500} hasMore button />)
      .setState({
        isLoading: false,
      });
    expect(wrapper).toMatchSnapshot();
  });

  it('should render <InfiniteScroll> with custom button', () => {
    const wrapper = enzyme
      .shallow(
        <InfiniteScroll
          loadItems={mockCallback}
          containerHeight={500}
          data={getData(10)}
          hasMore
          button={({ onClick }) => (
            <button onClick={onClick} type="button">
              more
            </button>
          )}
        />,
      )
      .setState({
        isLoading: false,
      });
    expect(wrapper).toMatchSnapshot();
  });

  it('should render <InfiniteScroll> without fixed height container', () => {
    const wrapper = enzyme
      .shallow(<InfiniteScroll loadItems={mockCallback} data={getData(10)} hasMore useWindow />)
      .setState({
        isLoading: false,
      });
    expect(wrapper).toMatchSnapshot();
  });
});
