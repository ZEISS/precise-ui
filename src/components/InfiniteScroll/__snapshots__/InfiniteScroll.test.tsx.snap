// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<InfiniteScroll /> should render <InfiniteScroll> with "Show More" button and data elements 1`] = `
<styled.div
  height={500}
  innerRef={[Function]}
>
  <div>
    <div
      key="X0"
    >
      X:
      X0
    </div>
    <div
      key="X1"
    >
      X:
      X1
    </div>
    <div
      key="X2"
    >
      X:
      X2
    </div>
    <div
      key="X3"
    >
      X:
      X3
    </div>
    <div
      key="X4"
    >
      X:
      X4
    </div>
    <div
      key="X5"
    >
      X:
      X5
    </div>
    <div
      key="X6"
    >
      X:
      X6
    </div>
    <div
      key="X7"
    >
      X:
      X7
    </div>
    <div
      key="X8"
    >
      X:
      X8
    </div>
    <div
      key="X9"
    >
      X:
      X9
    </div>
  </div>
  <styled.div>
    <Styled(Button)
      buttonStyle="secondary"
      icon="Add"
      onClick={[Function]}
      theme={
        Object {
          "buttonIconPosition": "left",
        }
      }
    >
      Show more
    </Styled(Button)>
  </styled.div>
</styled.div>
`;

exports[`<InfiniteScroll /> should render <InfiniteScroll> with custom button 1`] = `
<styled.div
  height={500}
  innerRef={[Function]}
>
  <div>
    <div
      key="X0"
    >
      X:
      X0
    </div>
    <div
      key="X1"
    >
      X:
      X1
    </div>
    <div
      key="X2"
    >
      X:
      X2
    </div>
    <div
      key="X3"
    >
      X:
      X3
    </div>
    <div
      key="X4"
    >
      X:
      X4
    </div>
    <div
      key="X5"
    >
      X:
      X5
    </div>
    <div
      key="X6"
    >
      X:
      X6
    </div>
    <div
      key="X7"
    >
      X:
      X7
    </div>
    <div
      key="X8"
    >
      X:
      X8
    </div>
    <div
      key="X9"
    >
      X:
      X9
    </div>
  </div>
  <styled.div>
    <button
      onClick={[Function]}
    />
  </styled.div>
</styled.div>
`;

exports[`<InfiniteScroll /> should render <InfiniteScroll> without fixed height container 1`] = `
<styled.div
  innerRef={[Function]}
>
  <div>
    <div
      key="X0"
    >
      X:
      X0
    </div>
    <div
      key="X1"
    >
      X:
      X1
    </div>
    <div
      key="X2"
    >
      X:
      X2
    </div>
    <div
      key="X3"
    >
      X:
      X3
    </div>
    <div
      key="X4"
    >
      X:
      X4
    </div>
    <div
      key="X5"
    >
      X:
      X5
    </div>
    <div
      key="X6"
    >
      X:
      X6
    </div>
    <div
      key="X7"
    >
      X:
      X7
    </div>
    <div
      key="X8"
    >
      X:
      X8
    </div>
    <div
      key="X9"
    >
      X:
      X9
    </div>
  </div>
  <styled.div />
</styled.div>
`;

exports[`<InfiniteScroll /> should render an empty <InfiniteScroll> 1`] = `
<styled.div
  height={500}
  innerRef={[Function]}
>
  <div />
  <styled.div />
</styled.div>
`;
